src/Tooling/Bit.Tooling.Templates/TodoTemplate/Web/Pages/SignUp.razor.cs

```diff
@@ -60,14 +63,49 @@ await HttpClient.PostAsJsonAsync("Auth/SignUp", new()
         }
     }
 
+    private async Task ResendLink()
+    {
+        if (IsLoading)
+        {
+            return;
+        }
+
+        IsLoading = true;
+        SignUpMessage = null;
+
+        try
+        {
+            await HttpClient.PostAsJsonAsync("Auth/SendConfirmationEmail", new()
+            {
+                Email = SignUpModel.Email
+            }, TodoTemplateJsonContext.Default.SendConfirmationEmailRequestDto);
+
+            SignUpMessageType = BitMessageBarType.Success;
+
+            SignUpMessage = "The confirmation link has been re-sent.";
+        }
+        catch (KnownException e)
+        {
+            SignUpMessageType = BitMessageBarType.Error;
+
+            SignUpMessage = ErrorStrings.ResourceManager.Translate(e.Message, SignUpModel.Email!);
```

Comment:
plz don't use `!` to bypass nullability checks, instead check at the start of the method that this property (`SignUpModel.Email`) is not null (or if it is null, simply return).